src/Numerics/Mesh/Topologies.jl

```diff
@@ -735,7 +783,190 @@ function StackedBrickTopology(
     nb = length(bndytoelem)
 
     T = eltype(basetopo.elemtocoord)
-
+    #----setting up DSS----------
+    if basetopo.elemtouvert isa Array
+        DA = CUDA.has_cuda_gpu() ? CuArray : Array
```

Comment:
I don't think we want this. We want to query the clima array type. We want to be able to run on the CPU even if the machine has GPU.

I would recommend just making an `Array` and doing all the transfers the way we currently do (i.e., later in the run)

Reply:
revised.